monai/inferers/splitter.py

```diff
@@ -58,142 +91,370 @@ def __call__(self, inputs: Any) -> Iterable[tuple[torch.Tensor, Sequence[int]]]:
+    """Split the whole slide image input into patches with sliding window strategy and a possible overlap.
+    This extract patches from file without loading the entire slide into memory.
+    It also allows to offset the starting position and filter the patches.
+
+    Args:
+        patch_size : the size of the patches to be generated.
+        offset: the amount of offset for the patches with respect to the original input.  Defaults to 0.
+        overlap: the amount of overlap between patches in each dimension. It can be either a float in
+            the range of [0.0, 1.0) that defines relative overlap to the patch size, or it can be a non-negative int
+            that defines number of pixels for overlap. Defaults to 0.0.
+        filter_fn: a callable to filter patches. It should accepts exactly two parameters (patch, location), and
+            return True for a patch to keep. Defaults to no filtering.
+        pad_mode: define the mode for padding. Either "constant" or None. Default to None.
+            Padding is only supported with "OpenSlide" or "cuCIM" backend, and the filling value is 256.
+        device: the device where the patches are generated. Defaults to the device of inputs.
+        reader: the module to be used for loading whole slide imaging. If `reader` is
+
+            - a string, it defines the backend of `monai.data.WSIReader`. Defaults to cuCIM.
+            - a class (inherited from `BaseWSIReader`), it is initialized and set as wsi_reader.
+            - an instance of a class inherited from `BaseWSIReader`, it is set as the wsi_reader.
+
+        reader_kwargs: the arguments to pass to `WSIReader` or the provided whole slide reader class.
+            For instance, level=2, dtype=torch.float32, etc.
+            Note that if `level` is not provided, `level=0` is assumed.
+
+    Note:
+        When a scaler value is provided for `patch_size`, `offset`, or `overlap`,
+        it is broadcasted to all the spatial dimensions.
+    """
+
+    def __init__(
+        self,
+        patch_size: Sequence[int] | int,
+        overlap: Sequence[float] | float | Sequence[int] | int = 0.0,
+        offset: Sequence[int] | int = 0,
+        filter_fn: Callable | None = None,
+        pad_mode: str | None = "constant",
+        device: torch.device | str | None = None,
+        reader: str | BaseWSIReader | type[BaseWSIReader] | None = "OpenSlide",
+        **reader_kwargs: dict,
+    ) -> None:
+        if pad_mode and pad_mode != "constant":
+            raise ValueError(
+                f"The underlying wsi readers only support for constant padding. pad_mod={pad_mode} is given."
+            )
+
+        super().__init__(
+            patch_size=patch_size, overlap=overlap, offset=offset, filter_fn=filter_fn, device=device, pad_mode=pad_mode
+        )
+        # Set WSI reader
+        self._set_reader(reader, reader_kwargs)
+        if self.reader.backend.lower() == "tifffile":
```

Comment:
Maybe no need to handle `tifffile` reader particularly? I feel it's a little bit hacky.
Just describing the supported readers in docstring is enough?

Thanks.

Reply:
The problem is the users most probably does not know about inefficiency of TiffFile when dealing with extracting patches, so I believe at least a warning is essential. What I can do to make it better is to check the reader if the reader is not one of the supported/tested ones (cucim and openslide) and give a warning for the rest.

Reply:
OK, this alternative solution sounds good. Thanks.

Reply:
done!